router/core/context.go

```diff
@@ -253,11 +253,43 @@ type requestContext struct {
 	expressionContext expr.Context
 }
 
+func (c *requestContext) setErrorContext() {
+	if c.error == nil {
+		return
+	}
+	c.expressionContext.Request.Error = &ExprWrapError{c.error}
+}
+
+func (c *requestContext) ResolveAnyExpressionWithErrorOverride(expression *vm.Program, err error) (any, error) {
+	currContextRequest := c.expressionContext.Request
+	if err != nil {
+		// We don't really want to do a deep clone since we only know that error only can get modified
+		copyContext := expr.Context{
```

Comment:
I ran into a problem where when an error occurs in the subgraph, it's not assigned to the `requestContext.error` at the point of being called.

Thus any expressions like `request.error?? 'no-error'` returned `'no-error'`. Observing the same pattern that was used in the predefined context fields for `request_error` which passes an error directly, I used a similar approach / workaround. Creating a copy of the expressionContext (non deep) and assigning it to resolve the expression.